SpokestackTests/SpeechPipelineTest.swift

```diff
@@ -30,142 +31,195 @@ class SpeechPipelineTest: XCTestCase {
+        let config = SpeechConfiguration()
+        config.stages = []
+        let p = SpeechPipeline(configuration: config, listeners: [delegate])
         wait(for: [didInitExpectation], timeout: 1)
         
         // start and stop the pipeline
         delegate.asyncExpectation = didStartExpectation
         p.start()
         wait(for: [didStartExpectation], timeout: 1)
-        XCTAssert(!p.context.isActive)
+        XCTAssertFalse(p.context.isActive)
         delegate.asyncExpectation = didStopExpectation
         p.stop()
         wait(for: [didStopExpectation], timeout: 1)
-        sleep(5)
-        XCTAssert(!p.context.isActive)
+        XCTAssertFalse(p.context.isActive)
+    }
+    
+    /// integration test
+    func testSpeechProcessors() {
+        let didStartExpectation = expectation(description: "didStartExpectation fulfills when testSpeechProcessors calls SpeechPipelineTestDelegate as the result of didStart method completion")
+        let didStopExpectation = expectation(description: "didStopExpectation fulfills when testSpeechProcessors calls SpeechPipelineTestDelegate as the result of didStop method completion")
+        let delegate = SpeechPipelineTestDelegate()
+        let config = SpeechConfiguration()
+
+        // init the pipeline
+        let p = SpeechPipeline(configuration: config, listeners: [])
+        
+        // add stages
+        let processor = TestProcessor(true, config: config, context: p.context)
+        p.context.listeners = [delegate]
+        p.context.stageInstances = [processor]
+        delegate.asyncExpectation = didStartExpectation
+        p.start()
+        wait(for: [didStartExpectation], timeout: 1)
+        XCTAssert(p.context.isActive)
+        delegate.asyncExpectation = didStopExpectation
+        p.stop()
+        wait(for: [didStopExpectation], timeout: 1)
+        XCTAssertFalse(p.context.isActive)
     }
 }
 
-class SpeechPipelineTestDelegate: PipelineDelegate, SpeechEventListener {
+class SpeechPipelineBuilderTest: XCTestCase {
+    func testBuild() {
+        let delegate = SpeechPipelineTestDelegate()
+        let didInit1Expectation = expectation(description: "didInitExpectation fulfills when testBuild calls SpeechPipelineTestDelegate as the result of build method completion")
+
+        // tflite
+        delegate.asyncExpectation = didInit1Expectation
+        let expectedTFStages: [SpeechProcessors] = [.vad, .tfLiteWakeword, .appleSpeech]
+        let p1 = SpeechPipelineBuilder()
+            .useProfile(.tfLiteWakewordAppleSpeech)
+            .setListener(delegate)
+            .build()
+        wait(for: [didInit1Expectation], timeout: 1)
+        XCTAssertEqual(expectedTFStages, p1.configuration.stages)
+        XCTAssert(delegate === p1.context.listeners[0])
```

Comment:
`XCTAssertEqual` (here and below), or doesn't that do instance equality?

Reply:
the latter